llvm/test/MC/Disassembler/AMDGPU/gfx8_mimg_features.txt

```diff
@@ -192,10 +271,10 @@
 # VI: image_atomic_add v5, v1, s[8:15] dmask:0x2 unorm ; encoding: [0x00,0x12,0x48,0xf0,0x01,0x05,0x02,0x00]
 0x00,0x12,0x48,0xf0,0x01,0x05,0x02,0x00
 
-# VI: image_atomic_add v5, v1, s[8:15] dmask:0x7 unorm ; encoding: [0x00,0x17,0x48,0xf0,0x01,0x05,0x02,0x00]
+# VI: image_atomic_add v[5:7], v1, s[8:15] dmask:0x7 unorm ; encoding: [0x00,0x17,0x48,0xf0,0x01,0x05,0x02,0x00]
 0x00,0x17,0x48,0xf0,0x01,0x05,0x02,0x00
 
-# VI: image_atomic_add v[5:9], v1, s[8:15] dmask:0xf unorm ; encoding: [0x00,0x1f,0x48,0xf0,0x01,0x05,0x02,0x00]
+# VI: image_atomic_add v5, v1, s[8:15] dmask:0xf unorm ; encoding: [0x00,0x1f,0x48,0xf0,0x01,0x05,0x02,0x00]
```

Comment:
There are 4 bits set in dmask so why isn't this `v[5:8]`?

Reply:
Disassembler defaults to V1 variants. That would have to be `IMAGE_ATOMIC_ADD_V4_V*` but there are not `V4_V* `variants for `IMAGE_ATOMIC`s, only `V1`, `V2` (64 or 32+tfe) and `V3` (64+tfe). 

Only cmpswap needs forth variant and this patch deletes `V4`s for non cmpswap atomics.

Reply:
Mirko is right. For non-cmpswap atomics, dmask can only be 0x1 or 0x3. So the destination reg can be 96b at most (dmask=0x3 + tfe). In the above example, the binary is incorrect to begin with (because dmask=0xf), the disassemble result is also incorrect.

Reply:
Thanks for the explanation. It would still be nice to handle this better, either by printing the `v[5:8]` (maybe with a comment that something is wrong) or by diagnosing it properly as an error case.

Reply:
I'll look into this and maybe do a separate PR.